drivers/nrf_802154_sl/src/platform/clock/nrf_802154_clock_zephyr.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ *   This file implements the nrf 802.15.4 HF Clock abstraction with Zephyr API.
+ *
+ * This implementation uses Zephyr API for clock management.
+ */
+
+#include <platform/clock/nrf_802154_clock.h>
+
+#include <stddef.h>
+
+#include <compiler_abstraction.h>
+#include <drivers/clock_control/nrf_clock_control.h>
+#include <drivers/clock_control.h>
+
+static bool hfclk_is_running;
+static bool lfclk_is_running;
+
+void nrf_802154_clock_init(void)
+{
+	/* Intentionally empty. */
+}
+
+void nrf_802154_clock_deinit(void)
+{
+	/* Intentionally empty. */
+}
+
+static void hfclk_on_callback(struct device *dev, clock_control_subsys_t subsys,
+			      void *user_data)
+{
+	hfclk_is_running = true;
+	nrf_802154_clock_hfclk_ready();
+}
+
+void nrf_802154_clock_hfclk_start(void)
+{
+	static struct clock_control_async_data clk_data = {
+		.cb = hfclk_on_callback
+	};
+	struct device *clk;
+
+	clk = device_get_binding(DT_LABEL(DT_INST(0, nordic_nrf_clock)));
+	__ASSERT_NO_MSG(clk != NULL);
```

Comment:
File taken originally https://github.com/zephyrproject-rtos/hal_nordic/tree/c92baabf36b081886a0668929b32f4f4ac21d490, just changed license header as @rubos mentioned here https://github.com/nrfconnect/sdk-nrf/pull/2303#discussion_r429801093
Possibly upstream needs some improvements but I suppose it will happen later.